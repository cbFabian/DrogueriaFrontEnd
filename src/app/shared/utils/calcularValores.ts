export const calcularValores = (producto: string, 
    reunion: string, esFestivo: boolean) => {
        if(!producto && !reunion){
            return 0;
        }
        if(!producto){
            return JSON.parse(reunion).precio;
        }
        if(!reunion){
            return JSON.parse(producto).precio;
        }
        let valorTotal: number = JSON.parse(producto).precio + JSON.parse(reunion).precio;
        if(esFestivo){
            valorTotal *= 2;
        }                
        return valorTotal;
};
