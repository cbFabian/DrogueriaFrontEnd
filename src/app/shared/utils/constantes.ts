export const constantes = {
    HORAS24: 24,
    HORA15: 15,
    HORA19: 19,
    MINUTOS60: 60,
    SEGUNDOS60: 60,
    MILISEGUNDOS: 1000,
    POSICION_DOS: 2,
    POSICION_TRES: 3,
    POSICION_CUATRO: 4,
    NUMERO_DOS: 2,
    NUMERO_TRES: 3,
    HORAMINIMA: 4,
    NUMERO_CINCO: 5,
    HORAMAXIMA: 8,
};
