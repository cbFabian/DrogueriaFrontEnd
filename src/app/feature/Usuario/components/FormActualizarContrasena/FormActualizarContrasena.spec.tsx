import * as React from 'react';
import { RenderResult, fireEvent, render, wait } from '@testing-library/react';
import { SinonStub, stub } from 'sinon';
import { FormActualizarContrasena } from '.';
import { setTextEvent } from 'app/shared/utils/test';

const POSICION_DOS = 2;
const NUMERO_DOS = 2;
const NUMERO_TRES = 3;

describe('FormActualizarContrasena test', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof FormActualizarContrasena> & {
    onSubmit: SinonStub;
  };

  beforeEach(() => {
    componentProps = {
      formTitle: 'Form test',
      usuario: {nombre: 'Lorem', clave: '1234'},
      mensajeError: '',
      onSubmit: stub(),
      borrarMensajeError: stub(),
    };
    componentWrapper = render(<FormActualizarContrasena {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });

  it('should fail on submit all fields missing', async () => {
    const elem = componentWrapper.container;
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(NUMERO_TRES);
    expect(spans[0].textContent).toBe('El campo clave actual es requerido.');
    expect(spans[1].textContent).toBe('El campo de nueva clave es requerido.');
    expect(spans[POSICION_DOS].textContent).toBe('El campo de confirmar nueva clave es requerido.');
  });

  it('should fail on submit two fields missing', async () => {
    const elem = componentWrapper.container;
    const claveActual = elem.querySelector('input[name="claveActual"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
        claveActual && fireEvent.change(claveActual, setTextEvent('claveActual', '1234'));
    });
    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(NUMERO_DOS);
    expect(spans[0].textContent).toBe('El campo de nueva clave es requerido.');
    expect(spans[1].textContent).toBe('El campo de confirmar nueva clave es requerido.');
  });

  it('should fail on submit one fields missing', async () => {
    const elem = componentWrapper.container;

    const claveActual = elem.querySelector('input[name="claveActual"]');
    const claveNueva = elem.querySelector('input[name="claveNueva"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
        claveActual && fireEvent.change(claveActual, setTextEvent('claveActual', '1234'));
    });
    await wait(() => {
        claveNueva && fireEvent.change(claveNueva, setTextEvent('claveNueva', '4321'));
    });
    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(1);
    expect(spans[0].textContent).toBe('El campo de confirmar nueva clave es requerido.');
  });

  it('should fail on submit compare two fields claveNueva and confirmarclaveNueva', async () => {
    const elem = componentWrapper.container;

    const claveActual = elem.querySelector('input[name="claveActual"]');
    const claveNueva = elem.querySelector('input[name="claveNueva"]');
    const confirmarclaveNueva = elem.querySelector('input[name="confirmarclaveNueva"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
        claveActual && fireEvent.change(claveActual, setTextEvent('claveActual', '1234'));
    });
    await wait(() => {
        claveNueva && fireEvent.change(claveNueva, setTextEvent('claveNueva', '4321'));
    });
    await wait(() => {
      confirmarclaveNueva && fireEvent.change(confirmarclaveNueva, setTextEvent('confirmarclaveNueva', '4322'));
    });
    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const spans = elem.querySelectorAll('span');
    expect(spans.length).toBe(1);
    expect(spans[0].textContent).toBe('Error, La nueva clave no coincide');
  });

  it('should submit', async () => {
    const elem = componentWrapper.container;

    const claveActual = elem.querySelector('input[name="claveActual"]');
    const claveNueva = elem.querySelector('input[name="claveNueva"]');
    const confirmarclaveNueva = elem.querySelector('input[name="confirmarclaveNueva"]');
    const submitButton = elem.querySelector('button[type="submit"]');

    await wait(() => {
        claveActual && fireEvent.change(claveActual, setTextEvent('claveActual', '1234'));
    });
    await wait(() => {
        claveNueva && fireEvent.change(claveNueva, setTextEvent('claveNueva', '4321'));
    });
    await wait(() => {
        confirmarclaveNueva && fireEvent.change(confirmarclaveNueva, setTextEvent('confirmarclaveNueva', '4321'));
    });
    await wait(() => {
      submitButton && fireEvent.click(submitButton);
    });
    const formSubmitted = componentProps.onSubmit.firstCall.args[0];
    expect(formSubmitted.nombre).toBe('Lorem');
    expect(formSubmitted.claveActual).toBe('1234');
    expect(formSubmitted.claveNueva).toBe('4321');
  });
});
