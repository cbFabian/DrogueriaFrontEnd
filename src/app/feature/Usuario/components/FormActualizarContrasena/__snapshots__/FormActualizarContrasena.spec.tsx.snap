// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormActualizarContrasena test should match snapshot 1`] = `
<div>
  <form>
    <h2>
      Form test
    </h2>
    <input
      class="sc-AxirZ eAukyj"
      name="claveActual"
      placeholder="Clave Actual"
      value=""
    />
    <input
      class="sc-AxirZ eAukyj"
      name="claveNueva"
      placeholder="Ingrese la nueva Clave"
      value=""
    />
    <input
      class="sc-AxirZ eAukyj"
      name="confirmarclaveNueva"
      placeholder="Confirme la Nueva Clave"
      value=""
    />
    <button
      class="sc-AxjAm kgxTKy"
      type="submit"
    >
      Actualizar
    </button>
  </form>
</div>
`;
